### ISSUE TITLE
Research: support using evalml components in sklearn pipelines

### ISSUE THREAD

**Issue by dsherry** [alteryx/evalml#859]
**Background**
@SydneyAyx @roland-hochmuth @piguy314159265 @kmax12 and @dsherry had a discussion today about the compatibility of pipeline datastructures between ML libraries like evalml and sklearn.

Evalml components were designed to be used within evalml pipelines. This issue tracks updating evalml components so they can be used within sklearn pipelines.

Notes with more context available [here](https://alteryx.quip.com/MqsbADmAzXqI/Using-EvalML-Components-with-SKLearn-Pipelines-Notes).

**Next steps**
How much work is required to be able to use evalml components in sklearn pipelines? Do a code spike and document, in order of priority

* What changes are necessary in evalml components for support of basic functionality (init/fit/predict).
* What's needed to obtain *identical* numerical results from evalml pipelines and sklearn pipelines, when trained and evaluated on the same datasets and with the same random seed.
* What, if any, advanced sklearn pipeline features aren't supported using evalml components today, and an idea of what would be required to support each. Example: miscellaneous APIs like feature importance.
* Ideas appreciated on what sort of test coverage would need to be added to fully support this use-case in evalml.

If the work required for this support isn't prohibitive, we should move forward. Otherwise we'll reevaluate.

Timeline: @SydneyAyx et al need this information by late August. Adding to the July milestone.
